src/backends/pgfplotsx.jl

```diff
@@ -0,0 +1,844 @@
+using Contour: Contour
+Base.@kwdef mutable struct PGFPlotsXPlot
+    is_created::Bool = false
+    was_shown::Bool = false
+    the_plot::PGFPlotsX.TikzDocument = PGFPlotsX.TikzDocument()
+    function PGFPlotsXPlot(is_created, was_shown, the_plot)
+        pgfx_plot = new(is_created, was_shown, the_plot)
+        # tikz libraries
+        PGFPlotsX.push_preamble!(pgfx_plot.the_plot, "\\usetikzlibrary{arrows.meta}")
+        PGFPlotsX.push_preamble!(pgfx_plot.the_plot, "\\usetikzlibrary{backgrounds}")
+        PGFPlotsX.push_preamble!(pgfx_plot.the_plot,
+        """
+        \\pgfkeys{/tikz/.cd,
+          background color/.initial=white,
+          background color/.get=\\backcol,
+          background color/.store in=\\backcol,
+        }
+        \\tikzset{background rectangle/.style={
+            fill=\\backcol,
+          },
+          use background/.style={
+            show background rectangle
+          }
+        }
+        """
+        )
+        # pgfplots libraries
+        PGFPlotsX.push_preamble!(pgfx_plot.the_plot, "\\usepgfplotslibrary{patchplots}")
+        PGFPlotsX.push_preamble!(pgfx_plot.the_plot, "\\usepgfplotslibrary{fillbetween}")
+        pgfx_plot
+    end
+end
+
+## end user utility functions
+function pgfx_axes(pgfx_plot::PGFPlotsXPlot)
+    gp =  pgfx_plot.the_plot.elements[1].elements[1]
+    return gp isa PGFPlotsX.GroupPlot ? gp.contents : gp
+end
+
+function pgfx_preamble(pgfx_plot::Plot{PGFPlotsXBackend})
+    old_flag = pgfx_plot.attr[:tex_output_standalone]
+    pgfx_plot.attr[:tex_output_standalone] = true
+    fulltext = String(repr("application/x-tex", pgfx_plot))
+    preamble = fulltext[1:first(findfirst("\\begin{document}", fulltext)) - 1]
+    pgfx_plot.attr[:tex_output_standalone] = old_flag
+    preamble
+end
+##
+
+function surface_to_vecs(x::AVec, y::AVec, s::Union{AMat, Surface})
+    a = Array(s)
+    xn = Vector{eltype(x)}(undef, length(a))
+    yn = Vector{eltype(y)}(undef, length(a))
+    zn = Vector{eltype(s)}(undef, length(a))
+    for (n, (i, j)) in enumerate(Tuple.(CartesianIndices(a)))
+        xn[n] = x[j]
+        yn[n] = y[i]
+        zn[n] = a[i,j]
+    end
+    return xn, yn, zn
+end
+
+function Base.push!(pgfx_plot::PGFPlotsXPlot, item)
+    push!(pgfx_plot.the_plot, item)
+end
+
+function (pgfx_plot::PGFPlotsXPlot)(plt::Plot{PGFPlotsXBackend})
+    if !pgfx_plot.is_created
+        the_plot = PGFPlotsX.TikzPicture(PGFPlotsX.Options())
+        rows, cols = size(plt.layout.grid)
+        bgc = plt.attr[:background_color_outside] == :match ? plt.attr[:background_color] : plt.attr[:background_color]
```

Comment:
This conditional picks between two identical alternatives.

Reply:
I'm guessing that it should be
```julia 
bgc = plt.attr[:background_color_outside] == :match ? plt.attr[:background_color] : plt.attr[:background_color_outside] 
```

Reply:
you are right, thanks for pointing out.